src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1318,61 +1318,93 @@ public override BoundNode VisitConditionalOperator(BoundConditionalOperator node
             VisitCondition(node.Condition);
             var consequenceState = this.StateWhenTrue;
             var alternativeState = this.StateWhenFalse;
+
+            var consequence = VisitConditionalOperand(consequenceState, node.Consequence, isByRef);
+            Unsplit();
+            consequenceState = this.State;
+            var consequenceResult = _result;
+
+            var alternative = VisitConditionalOperand(alternativeState, node.Alternative, isByRef);
+            Unsplit();
+            var alternativeResult = _result;
+
+            bool? getIsNullable(Result result) => (object)result.Type == null ? true : result.Type.IsNullable;
```

Comment:
Perhaps `result.Type is null ? ...`.
Feels like there is a condensed expression to do the job as well, but I actually prefer the explicit conditional ;-)